internal/security/config/command/proxy/adduser/command.go

```diff
@@ -0,0 +1,340 @@
+	flagSet.StringVar(&cmd.username, "user", "", "Username of the user to add")
+	flagSet.StringVar(&cmd.group, "group", "admin", "Group to which the user belongs, defaults to 'admin'")
+
+	flagSet.StringVar(&cmd.algorithm, "algorithm", "", "Algorithm used for signing the JWT, RS256 or ES256")
+	flagSet.StringVar(&cmd.publicKeyPath, "public_key", "", "Public key (in PEM format) used to validate the JWT.")
+	flagSet.StringVar(&cmd.jwtID, "id", "", "ID to use for linkage with JWT claim (usually the 'iss' field)")
+
+	flagSet.StringVar(&cmd.clientID, "client_id", "", "Optional manually-specified OAuth2 client_id.")
+	flagSet.StringVar(&cmd.clientSecret, "client_secret", "", "Optional manually-specified OAuth2 client_secret.")
+	flagSet.StringVar(&cmd.redirectUris, "redirect_uris", "", "OAuth2 redirect URL for browser-based users (required)")
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+	if cmd.tokenType == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --token-type is required", os.Args[0])
+	}
+	if cmd.tokenType != interfaces.JwtTokenType && cmd.tokenType != interfaces.OAuth2TokenType {
+		return nil, fmt.Errorf("%s proxy adduser: argument --token-type must be either 'jwt' or 'oauth2'", os.Args[0])
+	}
+	if cmd.username == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --user is required", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.JwtTokenType && cmd.algorithm == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --algorithm is required", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.JwtTokenType && cmd.algorithm != "RS256" && cmd.algorithm != "ES256" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --algorithm must be either 'RS256' or 'ES256'", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.JwtTokenType && cmd.publicKeyPath == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --public_key is required", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.OAuth2TokenType && cmd.redirectUris == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --redirect_uris is required", os.Args[0])
+	}
+
+	return &cmd, err
+}
+
+func (c *cmd) Execute() (statusCode int, err error) {
+	switch c.tokenType {
+	case interfaces.JwtTokenType:
+		statusCode, err = c.ExecuteAddJwt()
+	case interfaces.OAuth2TokenType:
+		statusCode, err = c.ExecuteAddOAuth2()
+	default:
+		statusCode = interfaces.StatusCodeExitWithError
+		err = fmt.Errorf("unsupported token type %s", c.tokenType)
+	}
+
+	return
+}
+
+func (c *cmd) createConsumer() error {
+	// Create kong consumer with the specified username
+	// https://docs.konghq.com/hub/kong-inc/jwt/#create-a-consumer
+
+	form := url.Values{
+		"username": []string{c.username},
+	}
+	kongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(), "consumers"}, "/")
+	c.loggingClient.Info(fmt.Sprintf("creating consumer (user) on the endpoint of %s", kongURL))
```

Comment:
It seems odd for a command-line tool to be logging informational log messages vs. printing this type of message to stdout in response to some sort of verbose command-line option...

Reply:
@lenny-intel Please weigh in.

Reply:
I get it, but since we are using the edgex boot strapping we end up with our standard logger controlled by the log level which in this case defaults to ERROR. Note that the bootstrap code will produce Info message when level is set to Info.

Reply:
This has been resolved by latest g-mod-bootstrap and setting the log level to ERROR.